src/System.ComponentModel.Annotations/tests/MaxLengthAttributeTests.cs

```diff
@@ -58,6 +86,16 @@ public static void Ctor_Int(int length)
         }
 
         [Theory]
+        [MemberData(nameof(ValidValues_ICollection))]
+        [MemberData(nameof(InvalidValues_ICollection))]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "MaxLengthAttribute in the full .NET framework doesn't support ICollection.Count. See https://github.com/dotnet/corefx/pull/2650")]
+        public void Validate_ICollection_NetFx_ThrowsInvalidCastException(MaxLengthAttribute attribute, object value)
```

Comment:
So what I meant is that we should have to tests (one for netcoreapp and one for netfx) and split the data Enumerable into to different data sets instead of constructing this data sets in runtime depending on the target that we are running on. While building this sets in runtime it makes it confusing to know which data sets work on netcoreapp and which ones on netfx.

See: https://github.com/dotnet/corefx/pull/17618

Does this clarify my comment?